### PR TITLE
ts/dpo-indicator

### PR DESCRIPTION
**Pull request by madepiet**
Added initial class pattern to dpo indicator.

### CODE REVIEW

File: ts/Stock/Indicators/DPO/DPOIndicator.ts [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+
+/* *
+*
+*   Prototype Properties
+*
+* */
+
+interface DPOIndicator {
+    nameBase: string;
+    options: DPOOptions;
```
Comment (bre1470) [inline ts/Stock/Indicators/DPO/DPOIndicator.ts:192]: The options property should go into the class pattern.

**Review by bre1470 — CHANGES_REQUESTED** [review]
Thank you! One small change needed, otherwise it is good.

File: ts/Stock/Indicators/DPOIndicator.ts [hunk 2]
```diff
@@ -1,217 +0,0 @@
...
-             * Period for Detrended Price Oscillator
-             */
-            period: 21
-        }
-    },
-    /**
-     * @lends Highcharts.Series#
-     */
-    {
-        nameBase: 'DPO',
```
Comment (bre1470) [inline ts/Stock/Indicators/DPOIndicator.ts:136]: I believe that the value of nameBase somehow got removed. This part needs to be restored with the `extend` call right under the interface of the prototype properties.
```ts
extend(DPOIndicator.prototype, {
    nameBase: 'DPO'
});
```

**Review by bre1470 — CHANGES_REQUESTED** [review]
Thank you so far. I found an other issue. Please see the comment.

Note that you also have to fix the auto merge conflict caused by changes of other indicators.

**Review by bre1470 — APPROVED** [review]
Thank you! Looks good. 👍

**Review by sebastianbochan — APPROVED** [review]

**Pull request merged**